pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,517 @@
+
+    >>> grouped_coeffs = [coeffs[list(indices)] for indices in H.grouping_indices]
+    >>> grouped_obs = [[H.ops[i] for i in indices] for indices in H.grouping_indices]
+    >>> grouped_coeffs
+    [array([1., 2.]), array([3.])]
+    >>> grouped_obs
+    [[X(0), X(1)], [Z(0)]]
+
+    Devices that evaluate a ``LinearCombination`` expectation by splitting it into its local observables can
+    use this information to reduce the number of circuits evaluated.
+
+    Note that one can compute the ``grouping_indices`` for an already initialized ``LinearCombination`` by
+    using the :func:`compute_grouping <pennylane.ops.LinearCombination.compute_grouping>` method.
+    """
+
+    num_wires = qml.operation.AnyWires
+    grad_method = "A"  # supports analytic gradients
+    batch_size = None
+    ndim_params = None  # could be (0,) * len(coeffs), but it is not needed. Define at class-level
+
+    def _flatten(self):
+        # note that we are unable to restore grouping type or method without creating new properties
+        return (self._coeffs, self._ops, self.data), (self.grouping_indices,)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        new_op = cls(data[0], data[1])
+        new_op._grouping_indices = metadata[0]  # pylint: disable=protected-access
+        new_op.data = data[2]
+        return new_op
+
+    def __init__(
+        self,
+        coeffs,
+        observables: List[Operator],
+        simplify=False,
+        grouping_type=None,
+        method="rlf",
+        _pauli_rep=None,
+        id=None,
+    ):
+        if qml.math.shape(coeffs)[0] != len(observables):
+            raise ValueError(
+                "Could not create valid LinearCombination; "
+                "number of coefficients and operators does not match."
+            )
+
+        self._coeffs = coeffs
+
+        self._ops = [convert_to_opmath(op) for op in observables]
+
+        self._hyperparameters = {"ops": self._ops}
+
+        self._grouping_indices = None
+
+        with qml.QueuingManager().stop_recording():
+            operands = [qml.s_prod(c, op) for c, op in zip(coeffs, observables)]
+
+        super().__init__(
+            *operands, grouping_type=grouping_type, method=method, id=id, _pauli_rep=_pauli_rep
+        )
+
+        if simplify:
+            # TODO clean up this logic, seems unnecesssarily complicated
+
+            simplified_coeffs, simplified_ops, pr = self._simplify_coeffs_ops()
+
+            self._coeffs = (
+                simplified_coeffs  # Losing gradient in case of torch interface at this point
+            )
+
+            self._ops = simplified_ops
+            with qml.QueuingManager().stop_recording():
+                operands = [qml.s_prod(c, op) for c, op in zip(self._coeffs, self._ops)]
+
+            super().__init__(*operands, id=id, _pauli_rep=pr)
+
+    def _check_batching(self):
+        """Override for LinearCombination, batching is not yet supported."""
```

Comment:
Do we have a story to add batching support for `LinearCombation`? We have it for the other operator arithmetic operators and it has been requested by external partners to support this type of thing.

Reply:
What is required for that? Should it work straight out of the box since `LinearCombination` is a `Sum`?

https://app.shortcut.com/xanaduai/story/59353

Reply:
```
>>> ([2,0] * qml.X(0) + qml.Y(1)).batch_size
2
```

Reply:
Added to https://app.shortcut.com/xanaduai/story/59353/upgrade-linearcombination and marking as resolved here